src/openrct2-ui/interface/Graph.cpp

```diff
@@ -182,10 +183,129 @@ static void graph_draw_line_b_money32(
     }
 }
 
+struct financial_tooltip_info
+{
+    const CoordsXY coords;
+    const money32 money{};
+};
+
+struct Rect
+{
+    const CoordsXY LeftTop;
+    const CoordsXY RightBottom;
+
+    int32_t GetLeft() const
+    {
+        return LeftTop.x;
+    }
+    int32_t GetTop() const
+    {
+        return LeftTop.y;
+    }
+    int32_t GetRight() const
+    {
+        return RightBottom.x;
+    }
+    int32_t GetBottom() const
+    {
+        return RightBottom.y;
+    }
+    int32_t GetWidth() const
+    {
+        return RightBottom.x - LeftTop.x;
+    }
+    int32_t GetHeight() const
+    {
+        return RightBottom.y - LeftTop.y;
+    }
+    bool Contains(const CoordsXY& coords) const
+    {
+        return coords.x >= LeftTop.x && coords.x <= RightBottom.x && coords.y >= LeftTop.y && coords.y <= RightBottom.y;
+    }
+    bool Contains(const ScreenCoordsXY& coords) const
+    {
+        return Contains(CoordsXY(coords.x, coords.y));
+    }
+};
+
+static constexpr auto chartMaxWidth = 63 * 6;
+static constexpr auto chartMaxHeight = 164;
+
+static int32_t indexForCursorAndHistory(const int32_t& historyCount, const int32_t& cursorX, const int32_t& chartX)
```

Comment:
No need for `const &` on `int32_t` and primitives smaller than integers.

Reply:
I was wondering about the convention for this 👍

Reply:
It probably doesn't matter after all compiler optimisation has taken place but avoid `const &` for primitives but always do it for structs. Any struct that fits in a register should be optimised by compiler like so otherwise you always force a copy to happen which if can be expensive depending on the copy constructor of the type.

Reply:
A very simple explanation:

A reference `&` is inherently a pointer.
Most primitives `bool`, `int`, `float` and etc will be the size of the pointer or smaller, so copying them or copying the pointer is kind of the same. For those, it's better to just copy them altogether.

Classes and structs, on the other hand, will typically contain multiple variables which makes using the reference a lot better for performance reasons, as you avoid copying stuff.

Reply:
@IntelOrca it does matter for compiler optimisation. Taking a reference forces instantiation on stack, passing by value lets compiler work its magic.

Verify it yourself: https://godbolt.org/z/AWr_iM

Reply:
Sounds good. I made the variables `const` since they don't need to be modified.

Reply:
@janisozaur When I said after "all" compiler optimisation, I am including link time optimisation. Try calling `bar` from `foo` and you will see it optimise.